src/lib/Form/Type/RelationType.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\ContentForms\Form\Type;
+
+use Ibexa\ContentForms\Form\Transformer\RelationTransformer;
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class RelationType extends AbstractType
+{
+    public const LOCATION_DEFAULT = 0;
+    public const LOCATION_BROWSE = 1;
+    public const LOCATION_SELF = -1;
+
+    private LocationService $locationService;
+
+    private TranslatorInterface $translator;
+
+    public function __construct(
+        LocationService $locationService,
+        TranslatorInterface $translator
+    ) {
+        $this->locationService = $locationService;
+        $this->translator = $translator;
+    }
+
+    public function getBlockPrefix()
+    {
+        return 'ibexa_form_type_relation';
+    }
+
+    public function buildForm(
+        FormBuilderInterface $builder,
+        array $options
+    ) {
+        $builder
+            ->add('location', LocationType::class)
+            ->add('location_type', LocationChoiceType::class, [
+                'multiple' => false,
+                'expanded' => true,
+                'label' => /** @Desc("Select Starting Location") */ 'content_forms.relation.location_type',
+                'choices' => [
+                    /** @Desc("Default") */
+                    'content_forms.relation.location_type.default' => self::LOCATION_DEFAULT,
+                    /** @Desc("Browse") */
+                    'content_forms.relation.location_type.browse' => self::LOCATION_BROWSE,
+                    /** @Desc("Self") */
+                    'content_forms.relation.location_type.self' => self::LOCATION_SELF,
+                ],
+            ]);
+
+        $builder->addModelTransformer(new RelationTransformer());
+    }
+
+    public function buildView(
+        FormView $view,
+        FormInterface $form,
+        array $options
+    ) {
+        $view->vars['destination_location'] = null;
+
+        if ($view->vars['value']) {
```

Comment:
I'm not 100% sure about this, but shouldn't we use `$form->getData()` here? Seems more apropriate to use that instead of a view-related value.

Additionally, it seems that form value is an array (as it contains two fields and no `data_class`). Will this really be correct?

Reply:
The field value is an integer if not empty. `RelationTransformer` ensures it for the BC sake.